### PR TITLE
Run pal2img tool in extract_agg scripts, minor code cleanup

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Related to #6553

### CODE REVIEW

File: src/engine/image_tool.cpp [hunk 1]
```diff
@@ -48,10 +49,10 @@
 
 namespace
 {
-    bool isPNGFilePath( const std::string & path )
+    bool isPNGFilePath( const std::string_view path )
     {
         const std::string pngExtension( ".png" );
-        return path.size() > pngExtension.size() && ( path.compare( path.size() - pngExtension.size(), pngExtension.size(), pngExtension ) == 0 );
+        return path.size() >= pngExtension.size() && ( path.compare( path.size() - pngExtension.size(), pngExtension.size(), pngExtension ) == 0 );
```
Comment (oleg-derevenetz) [inline src/engine/image_tool.cpp:55]: File name `.png` is a perfectly valid name of a file with extension "png" :)

File: src/tools/pal2img.cpp [hunk 2]
```diff
@@ -36,7 +36,7 @@ int main( int argc, char ** argv )
         std::string baseName = System::GetBasename( argv[0] );
 
         std::cerr << baseName << " generates an image with colors based on a provided palette file." << std::endl
-                  << "Syntax: " << baseName << " palette_file.pal output.[bmp|png]" << std::endl;
+                  << "Syntax: " << baseName << " palette_file.pal output.bmp|output.png" << std::endl;
```
Comment (oleg-derevenetz) [inline src/tools/pal2img.cpp:39]: If we think about it, square brackets imply optionality, which is not very suitable here, so my previous syntax recommendation was wrong.

File: src/engine/image_tool.cpp (hunk 1 above)
Comment (ihhub) [reply src/engine/image_tool.cpp:55]: No comments :)

**Review by ihhub — APPROVED** [review]
Hi @oleg-derevenetz , everything looks good. I left just one optional recommendation. Could you please check it?

File: script/agg/extract_agg.bat [hunk 3]
```diff
@@ -38,5 +38,6 @@ exit /B %EXIT_CODE%
 extractor agg *.AGG *.agg || exit /B 1
 icn2img icn\HEROES2 agg\HEROES2\kb.pal agg\HEROES2\*.icn || exit /B 1
 icn2img icn\HEROES2X agg\HEROES2\kb.pal agg\HEROES2X\*.icn || exit /B 1
+pal2img agg\HEROES2\kb.pal kb.png || exit /B 1
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply script/agg/extract_agg.bat]: Yes, it will look more understandable :)

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you so much for these improvements!

**Pull request merged**